test/unittest/rtps/history/HistoryTests.cpp

```diff
@@ -0,0 +1,255 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <fastdds/rtps/history/History.h>
+#include <fastdds/rtps/common/CacheChange.h>
+
+#include <tuple>
+
+using namespace eprosima::fastrtps::rtps;
+using namespace ::testing;
+using namespace std;
+
+class HistoryTests : public TestWithParam<tuple<int32_t, int32_t, uint32_t, MemoryManagementPolicy_t> >
+{
+protected:
+
+    class HistoryForTest : public History
+    {
+    public:
+        explicit HistoryForTest(
+                const HistoryAttributes& att)
+            : History(att)
+        {
+            mp_mutex = &mutex_;
+        }
+
+        virtual ~HistoryForTest()
+        {
+        }
+
+        bool remove_change(
+                CacheChange_t* ch) override
+        {
+            if (ch == nullptr)
+            {
+                return false;
+            }
+
+            std::lock_guard<eprosima::fastrtps::RecursiveTimedMutex> guard(*mp_mutex);
+            auto it = std::remove(m_changes.begin(), m_changes.end(), ch);
+            if (it == m_changes.end())
+            {
+                return false;
+            }
+
+            m_changes.erase(it, m_changes.end());
+            return true;
+        }
+
+    private:
+        eprosima::fastrtps::RecursiveTimedMutex mutex_;
+    };
+
+    HistoryTests()
+        : pool(nullptr)
+        , pool_size(10)
+        , max_pool_size(0)
+        , payload_size(128)
+        , memory_policy(MemoryManagementPolicy_t::PREALLOCATED_MEMORY_MODE)
+    {
+    }
+
+    virtual ~HistoryTests()
+    {
+    }
+
+    virtual void SetUp()
+    {
+        pool_size     = get<0>(GetParam());
+        max_pool_size = get<1>(GetParam());
+        payload_size  = get<2>(GetParam());
+        memory_policy = get<3>(GetParam());
+
+        HistoryAttributes att;
+        att.initialReservedCaches = pool_size;
+        att.maximumReservedCaches = max_pool_size;
+        att.payloadMaxSize = payload_size;
+        att.memoryPolicy = memory_policy;
+
+        pool = new HistoryForTest(att);
+    }
+
+    virtual void TearDown()
+    {
+        delete pool;
+    }
+
+    HistoryForTest* pool;
```

Comment:
Can we name this `history` or something similar?  Wen referencing this instance, I was reading it as one of the pools (CacheChange or Payload) and was confused.